### PR TITLE
contrib: update nix flake to use flake-utils

### PR DESCRIPTION
**Pull request by mjlbach**
For nix, expands the flake to operate on all supported operating systems. In nix-community we'll be using this to provide binary builds for mac and linux.

### CODE REVIEW

File: contrib/flake.nix [hunk 1]
```diff
@@ -3,93 +3,91 @@
...
+        };
+
+        defaultApp = apps.nvim;
+
+      }
+    );
+    # overlay = final: prev: {
+    #   inherit (self.packages."${system}") neovim neovim-debug;
+    # };
+    # devShell."${system}" = self.packages."${system}".neovim-developer;
```
Comment (teto) [inline contrib/flake.nix:92]: why remove my devShell :'( ?

File: contrib/flake.nix [hunk 2]
```diff
@@ -3,93 +3,91 @@
...
-      nvim-debug = mkApp self.packages."${system}".neovim-debug;
-    };
-
-    defaultApp."${system}" = self.apps."${system}".nvim;
-
-    devShell."${system}" = self.packages."${system}".neovim-developer;
-  };
+      rec {
+
+        packages = flake-utils.lib.flattenTree rec {
```
Comment (teto) [inline contrib/flake.nix:16]: I dont think you need flattenTreee, do you ?

File: contrib/flake.nix (hunk 1 above)
Comment (mjlbach) [reply contrib/flake.nix:92]: Don't worry it's coming back! That's why I marked it a draft

File: contrib/flake.nix (hunk 2 above)
Comment (mjlbach) [reply contrib/flake.nix:16]: I don't think so

**Comment by mjlbach** [comment]
@teto I guess we should discuss what to do/where to put this, everything works now but asan/ubsan on darwin which I think is a stdenv issue.

File: contrib/flake.nix [hunk 3]
```diff
@@ -3,93 +3,100 @@
```
Comment (teto) [inline contrib/flake.nix:3]: we can expand on the definition, something like "provides neovim nightly version for both users and developers"

**Pull request merged**